docs/source-pytorch/common/trainer.rst

```diff
@@ -1232,23 +1232,27 @@ reload_dataloaders_every_n_epochs
 
 |
 
-Set to a positive integer to reload dataloaders every n epochs.
+Set to a positive integer to reload dataloaders every n epochs from your currently used data source.
+DataSource can be a ``LightningModule`` or a ``LightningDataModule.
+
 
 .. code-block:: python
 
     # if 0 (default)
-    train_loader = model.train_dataloader()
+    train_loader = current_datasource.train_dataloader()
```

Comment:
I think this should stay concrete. You can put a comment line below like this
```py
train_loader = model.train_dataloader()
# or if using data module: datamodule.train_dataloader()
```

or similar.